pkg/serving/v1/gitops_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2020 The Knative Authors
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"gotest.tools/assert"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	servingtest "knative.dev/serving/pkg/testing/v1"
+
+	libtest "knative.dev/client/lib/test"
+	"knative.dev/pkg/ptr"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+)
+
+func TestGitOpsOperations(t *testing.T) {
+	// create clients
+	fooclient := NewKnServingGitOpsClient("foo-ns", "./tmp/kn/files")
```

Comment:
You could also go with `ioutils` tmp that will make sure it's temporary and unique dir, we have it used [here](https://github.com/knative/client/blob/master/pkg/kn/commands/service/create_test.go#L1012-L1013). 
Regardless of the actuall tmp dir creation,  adding `defer rm` might be beneficial here as well.

```
tempDir, err := ioutil.TempDir("", "kn-file")
defer os.RemoveAll(tempDir)
```

Reply:
yes, please use the `ioutils` package as this will work on Windows then, too.

Reply:
> yes, please use the `ioutils` package as this will work on Windows then, too.

That's a great point the paths should be constructed with `filepath.Join` to account for Windows eventually.

Reply:
have made this change